### PR TITLE
feat(sf_movies_2): POST /movies

### PR DESCRIPTION
**Pull request by vinay-pad**
What: Create and save new movies by POSTing to /movies endpoint

Why: These changes are part of the second part of how to build your own api, involving creating of database models and be able to store movies into the database.

Details:
This checkin involves a series of following steps followed in order to be able to create a movie object and store it in the database.
- Coming up with a database design
- Performing migrations to create movies table.
- Defining the movie model.
- Defining a serializer to tell the API how to serialize the movie object when returning it.
- Writing unit tests to test coverage of all the flows for the API.
- Writing validators to validate the input against the constraints defined in the model and writing corresponding validator unit tests.
- Write controller to take the input payload and save to DB and return the serialized model back.
- Writing integration test.

### CODE REVIEW

File: lib/validators/movie.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+'use strict';
```
Comment (thetonywu) [inline lib/validators/movie.js:1]: I know it doesn't really matter yet, but typically we use this convention for validators:

lib/validators/movies/create.js (this file)
lib/validators/movies/list.js (for listing movies, which you'll create later)

File: test/validators/movie.test.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+'use strict';
```
Comment (thetonywu) [inline test/validators/movie.test.js]: in line with the above comment, this file should become test/validators/movies/create.test.js

File: lib/plugins/features/movies/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const Controller     = require('./controller');
+const MovieValidator = require('../../../validators/movie');
```
Comment (thetonywu) [inline lib/plugins/features/movies/index.js]: MovieCreateValidator

**Review by thetonywu — COMMENTED** [review]
some minor comments

**Review by thetonywu — APPROVED** [review]
lgtm

**Comment by thetonywu** [comment]
I think a single LGTM is fine for this, feel free to merge!

**Comment by vinay-pad** [comment]
@tonywuhl thanks!

**Pull request merged**